akka-actor/src/main/scala/akka/actor/ActorRefProvider.scala

```diff
@@ -338,6 +338,9 @@ trait ActorRefFactory {
   /**
    * Stop the actor pointed to by the given [[akka.actor.ActorRef]]; this is
    * an asynchronous operation, i.e. involves a message send.
+   * If this method is applied to the `self` reference from inside an Actor
+   * then that Actor is guaranteed to not process any further messages before
```

Comment:
ah, I see :-) I’d still leave this in to make it absolutely clear that message dequeuing does not only stop after termination but _right now_.